src/ray/common/ray_syncer.h

```diff
@@ -0,0 +1,294 @@
+#pragma once
+#include <grpcpp/server.h>
+
+#include "absl/container/flat_hash_map.h"
+#include "absl/container/flat_hash_set.h"
+#include "ray/common/asio/instrumented_io_context.h"
+#include "ray/common/asio/periodical_runner.h"
+#include "ray/common/id.h"
+#include "src/ray/protobuf/ray_syncer.grpc.pb.h"
+
+namespace ray {
+namespace syncer {
+
+using ServerBidiReactor = grpc::ServerBidiReactor<ray::rpc::syncer::RaySyncMessages,
+                                                  ray::rpc::syncer::RaySyncMessages>;
+using ClientBidiReactor = grpc::ClientBidiReactor<ray::rpc::syncer::RaySyncMessages,
+                                                  ray::rpc::syncer::RaySyncMessages>;
+
+using ray::rpc::syncer::ClientMeta;
+using ray::rpc::syncer::Dummy;
+using ray::rpc::syncer::RayComponentId;
+using ray::rpc::syncer::RaySyncMessage;
+using ray::rpc::syncer::RaySyncMessages;
+using ray::rpc::syncer::RaySyncMessageType;
+using ray::rpc::syncer::ServerMeta;
+
+static constexpr size_t kComponentArraySize =
+    static_cast<size_t>(ray::rpc::syncer::RayComponentId_ARRAYSIZE);
+
+/// The interface for a reporter. Reporter is defined to be a local module which would
+/// like to let the other nodes know its status. For example, local cluster resource
+/// manager.
+struct Reporter {
```

Comment:
use `ReporterInterface` since it's a pure abstract class?

Reply:
maybe named it LocalReporterInterface for clarity.

Reply:
Then, I think then we should also rename the receiver to LocalReceiverInterface since it's also about the local view.

Reply:
Synced offline. I'll use `ReporterInterface`